modules/CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+# =====================================================================
+#               Modules Package Builder
+# =====================================================================
+# This CMake script:
+# 1. Reads module list from modules.dmd
+# 2. Downloads modules for target architecture using dmf-get
+# 3. Packages them using todmp
+# 4. Makes the package available for embedding in firmware
+# =====================================================================
+
+set(MODULES_DMD_FILE "${CMAKE_CURRENT_SOURCE_DIR}/modules.dmd")
+set(MODULES_DOWNLOAD_DIR "${CMAKE_CURRENT_BINARY_DIR}/dmf")
+set(MODULES_DMP_FILE "${CMAKE_CURRENT_BINARY_DIR}/modules.dmp")
+
+# Check if modules.dmd exists
+if(NOT EXISTS "${MODULES_DMD_FILE}")
+    message(STATUS "No modules.dmd file found, skipping modules package build")
+    return()
+endif()
+
+# Read and parse modules.dmd file
+file(STRINGS "${MODULES_DMD_FILE}" MODULES_LIST)
+
+# Filter out empty lines and comments
+set(MODULES_TO_FETCH "")
```

Comment:
Nie masz tego robić ręcznie - tak jak pisałem w tickecie jest do tego narzędzie `dmf-get` w obrazie dmod. Jeżeli go nie masz to sobie zbuduj dmod w trybie DMOD_SYSTEM i weź z tamtąd